### ISSUE TITLE
Duplicate Manager - Show Diff - Highlight characters --> highlights wrong characters

### ISSUE THREAD

**Issue by ThiloteE** [JabRef/jabref#9053]
JabRef 5.8 commit 0b580790d7a323ccdb445b8a8a6e8dd4d5917e69

**How to reproduce:**

Have two entries with groups:

A)  groups       = {test1, test2, test3},
B)  groups       = {test1, test4, test2, test3},

Reality:
![grafik](https://user-images.githubusercontent.com/73715071/184508028-43dda8d0-f691-450a-ad1e-8de350dd9528.png)

Expected:
![grafik](https://user-images.githubusercontent.com/73715071/184508113-c1d04e92-14ea-4e6e-a0d6-ebd2d1b3c826.png)

**Comment by HoussemNasri** [comment]
This can be done by changing the default diff separator to the groups separator `, ` when highlighting groups differences. This can be useful for highlighting keywords and files differences too.

https://github.com/JabRef/jabref/blob/bd77d7336fde1c4618f2e67dcf2ef22ceb1cda77/src/main/java/org/jabref/gui/mergeentries/newmergedialog/diffhighlighter/DiffHighlighter.java#L38-L40

**Comment by Anqiii-hub** [comment]
Hi I think I can fix this issue, could you assign this issue to me?

**Comment by ThiloteE** [comment]
Thank you :-)

As a general advice for newcomers: check out https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md for a start. Also, https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace is worth having a look at. Feel free to ask if you have any questions here on GitHub or also at JabRef's [Gitter](https://gitter.im/JabRef/jabref) chat.

Try to open a (draft) pull request early on, so that people can see you are working on the issue and so that they can see the direction the pull request is heading towards. This way, you will likely receive valuable feedback.

**Comment by ThiloteE** [comment]
@HoussemNasri Did you mean this preference?

![grafik](https://user-images.githubusercontent.com/73715071/196025608-3937336d-e8fa-45d5-9aff-1b43886706b2.png)

I think the diff separator should probably always be similar to what the user has set in this preference

**Comment by HoussemNasri** [comment]
No, they are totally different. The diff highlighter separator can be either a space or an empty string. It's used for splitting the text before comparison. For example for diffing `abcd w` and `ebdd sa`, we can split the texts by space and compare them: `abcd` vs `ebdd` and `w` vs `sa`, or we can compare them character by character *(split by an empty string)*: `a` vs `e`, `b` vs `b`, etc.

**Comment by Anqiii-hub** [comment]
> This can be done by changing the default diff separator to the groups separator `, ` when highlighting groups differences. This can be useful for highlighting keywords and files differences too.
> 
> https://github.com/JabRef/jabref/blob/bd77d7336fde1c4618f2e67dcf2ef22ceb1cda77/src/main/java/org/jabref/gui/mergeentries/newmergedialog/diffhighlighter/DiffHighlighter.java#L38-L40

Hi @ThiloteE,

- I followed the above method and created a pull request. I made the following changes：
- When calling showDiff method, I first check if the field is "Groups", if the field is Groups, I use "," as a seperator. I didn't change the seperator of other field. And the result shows as follows: <img width="1411" alt="Screen Shot 2022-10-23 at 8 34 29 pm" src="https://user-images.githubusercontent.com/83406036/197384873-608f3fbb-a367-496f-8aad-4dee1528a8ad.png">


- However, this method has its shortcoming, it only works when the user chooses to use "," to seperate different groups. Otherwise, the results will be like the following: <img width="1385" alt="Screen Shot 2022-10-23 at 8 36 02 pm" src="https://user-images.githubusercontent.com/83406036/197384942-eeac2552-8918-460b-8bb1-bfee7cf02120.png">


- To solve this bug, I also choose to check if the leftValue or rightValue contains(","), if neither of the value contains(","), I choose to still use "" as the seperator. The results is like the following, kind of solving the issue. <img width="1379" alt="Screen Shot 2022-10-23 at 8 52 17 pm" src="https://user-images.githubusercontent.com/83406036/197385556-7034d01c-1d6a-4304-aec8-ce9407e6e57a.png">


- @HoussemNasri said the seperator can also be used in the field "Keywords", however I think it‘s unnecessary. As the issue only occurs when each word has the same part (i.e "test"), the keywords field unlikely to have many words containing the same part. 


- However, I know this method doesn't solve the whold issue. I also tried if I could seperate the field value with "," first and then separate with ""; or use other methods in the com.github.difflib.DiffUtils (the library we use). Both of these two methods seem to need to change a lot of places and I failed. If you have any other suggestions, plz let me know.
